Upstream-Status: Backport

commit 5f5eb921765794e8fc58c4bdffa2daa2ae34800f
Author: James Youngman <jay@gnu.org>
Date:   Sat Feb 20 19:53:13 2010 +0000

    Fix Savannah bug#28872, Mistake in "Problems with -exec and filenames"
    
    * doc/find.texi (Problems with -exec and filenames): Add missing
    $0 argument in example for sh -c 'something "$@" sh ...
    * NEWS: Mention this change.
    
    Signed-off-by: James Youngman <jay@gnu.org>

diff --git a/ChangeLog b/ChangeLog
index 13539a4..e94ba96 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,10 @@
 2010-02-20  James Youngman  <jay@gnu.org>
 
+	Fix Savannah bug#28872, Mistake in "Problems with -exec and filenames"
+	* doc/find.texi (Problems with -exec and filenames): Add missing
+	$0 argument in example for sh -c 'something "$@" sh ...
+	* NEWS: Mention this change.
+
 	Fix Savannah bug# 28824: "-ctime x" yields "missing argument to
 	`-ctime'".
 	* find/parser.c (parse_fls): If the argument is invalid, reverse
diff --git a/NEWS b/NEWS
index 4e910df..4c97be9 100644
--- a/NEWS
+++ b/NEWS
@@ -4,6 +4,9 @@ GNU findutils NEWS - User visible changes.	-*- outline -*- (allout)
 
 ** Bug Fixes
 
+#28872: Mistake in "#safer" example in "Problems with -exec and
+        filenames" section of the Texinfo manual.
+
 #28824: Corrected error message for "-ctime x".
         Likewise for -gid, -inum, -links, -mmin, -cmin, -amin,
 	-uid, -used, -atime, -mtime, -ctime.
diff --git a/doc/find.texi b/doc/find.texi
index 2e5958d..391ffa0 100644
--- a/doc/find.texi
+++ b/doc/find.texi
@@ -4830,8 +4830,8 @@ problem:
 
 @example
 # safer
-find -exec sh -c 'something "$@@"' @{@} \;
-find -execdir sh -c 'something "$@@"' @{@}\;
+find -exec sh -c 'something "$@@"' sh @{@} \;
+find -execdir sh -c 'something "$@@"' sh @{@}\;
 @end example
 
 This approach is not guaranteed to avoid every problem, but it is much
